toolchain/lower/file_context.cpp

```diff
@@ -85,14 +86,22 @@ auto FileContext::Run() -> std::unique_ptr<llvm::Module> {
   for (auto [id, _] : sem_ir_->functions().enumerate()) {
     BuildFunctionDefinition(id);
   }
+
+  // Lower function definitions for generics.
+  // TODO: this needs to be an iterator as new definitions can be added
+  // while building.
```

Comment:
vector iterators aren't stable across insertions - so perhaps the opposite? The code you have looks like the right code for iterating through a vector that might have things added to it during the iteration? (because it re-queries the size each time, and indexes each time - rather than keeping an iterator/pointer or size from the start to the end))

Only one minor change: The bound check should be `!=` rather than `<` to match C++ norms.

Reply:
Yes, that's what I meant, but didn't use quite the right words :-). I updated the comment an an attempt to be more clear.